kasa/discover.py

```diff
@@ -147,7 +147,16 @@ class ConnectAttempt(NamedTuple):
     device: type
 
 
+class JsonDiscovered(NamedTuple):
+    """Try to connect attempt."""
+
+    ip: str
+    raw: dict
+    redactor: Callable[[dict], dict]
```

Comment:
I'm not sure if we really want to leak our internal implementation through this API. Would it be better just to pass the ip: raw data, and let the callback to use the payload as they want, i.e., in the case of our cli, we'd do redaction etc.?

Reply:
Removed and replaced with `ip` & `port` in `meta`